### DISCUSSION TITLE
Using eww with the i3 window manager.

### DISCUSSION THREAD

**Discussion by astrogewgaw** [elkowar/eww#380] [Q&A] [answered]
<div align="justify">

So I am using eww to make a simple powermenu, and I am using the i3 window manager. My window is named "powermenu" (surprise, surprise) and I tried starting it via a key binding in the i3 config but it is not working? It works via the terminal though (that is, `eww open powermenu` works). So, the question I have is: how should I open any eww widget via i3? Also, how do I get an eww widget to close via a key (like `ESC`)?

**NOTE**: I have the `eww` binary in my `PATH`.

</div>

**Comment by VuiMuich** [comment]
I don't think eww has any way to catch key events, but you can configure the window to be togglable so the same keybind that opens the window also closes it.
Theoretically it would be possible if you set the windowtype of your eww window to something that makes it 'focusable' for the WM you could try to add a hook to the ESC key events that if the focused window is `eww - powermenu` (not 100% sure about the `-` though) execute `eww close powermenu`.

Regarding the keybind not working, I believe this is most probably a little syntax hiccup in your i3conf.
Tbh my i3 days were quite some time ago but if Memphis serves right it should be something like `bindsym $mod+0 exec "eww open -toggle powerten"`

**Reply by astrogewgaw** [reply]
<div align="justify">

Well this didn't work. Firstly, the `eww` command does not have a `-toggle` option. Secondly, I have checked the syntax in my i3 configuration and it is correct. It just doesn't do anything. Not sure about the way you suggested to capture the `ESC` event though, since I cannot get the window open to try it out 😅 .

</div>

**Reply by VuiMuich** [reply]
Sorry, indeed it is `--toggle`. Was typing form top of my head while commuting, so I am sorry for this misleading info.

**Reply by astrogewgaw** [reply]
<div align="justify">

No worries 😁 ! I just checked the command out and it works! Made my work easier 👍🏾, I was using a shell script for no reason before to make the widget toggle. Should have checked the help for the `eww open` command 😅 .

</div>

**Comment by astrogewgaw** [comment, marked as answer]
<div align="justify">

Okay, solved it 😁 ! Apparently, `eww` was on my `PATH`, but i3 was not picking up on it. Giving it the absolute path for the `eww` binary worked. Just have to figure out how to make it toggle now.

</div>
